apps/web/src/middlewares/authHandler.ts

```diff
@@ -21,24 +21,28 @@ async function getUserOrUnauthorized() {
 
 export type HandlerFn<IReq extends DefaultParams, IResp extends object = {}> = (
   req: NextRequest,
-  context: AuthContext<IReq>,
+  res: NextResponse<IResp>,
+  context: Context<IReq>,
 ) => Promise<NextResponse<IResp>>
 
 export function authHandler<
   IReq extends DefaultParams,
   IResp extends object = {},
 >(handler: HandlerFn<IReq, IResp>) {
-  return async (req: NextRequest, { ...rest }: AuthContext<IReq>) => {
+  return async (
+    req: NextRequest,
+    res: NextResponse<IResp>,
+    _context: Context<IReq>,
+  ) => {
     const { user, workspace } = await getUserOrUnauthorized()
+    // This is Typescript lie, we know that res is NextResponse<IResp>
+    // But because we want the return signature of the handler to be
+    // (req: NextRequest, res: NextResponse<IResp>)
+    // So in runtime we know params is a Promise<IReq>
+    const requestContext = res as unknown as { params: Promise<IReq> }
+    const params = await requestContext.params
```

Comment:
[This docs](https://nextjs.org/docs/app/building-your-application/routing/route-handlers#dynamic-route-segments)